Geocoder/src/main/java/com/example/Geocoder/GeocoderController.java

```diff
@@ -1,25 +1,54 @@
 package com.example.Geocoder;
-
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
-import org.springframework.web.bind.annotation.GetMapping;
 
-import java.util.Map;
-import java.util.HashMap;
+import reactor.core.publisher.Mono;
 
 
 @RestController
+@RequestMapping("/api/address")
 class GeocoderController{
-    @GetMapping("/geocoder")
-    public Result getGeocode(){
-        Result result = new Result("Hello World", true, "Delievered");
-        return result;
+    private final CensusService censusService;
+
+    @Autowired //Injecting an instance of the CensusService
+    public GeocoderController(CensusService censusService){
+        this.censusService = censusService;
     }
 
-    @GetMapping("/geocoderMap")
-    public Map<String, String> getGeocodeMap(){
-        Map<String, String> result = new HashMap<>();
-        result.put("response", "Hello World");
-        result.put("status", "true");
-        return result;
+    
+    
+    // The response from the API call is processed and formatted into before being returned to the user. 
+    // Return response object
+    @PostMapping
+    public Mono<ResponseEntity<String>> submitAddress(@RequestBody Address address){
+        return censusService.submitAddress(address)
+            .flatMap(response ->  {
+                try {
+                    JSONObject jsonResponse = new JSONObject(response);
+                    JSONObject addressMatches = jsonResponse
+                        .getJSONObject("result")
+                        .getJSONArray("addressMatches")
+                        .getJSONObject(0);
+                    
+                    Response formattedResponse = new Response(
+                        addressMatches.getString("matchedAddress"), 
+                        addressMatches.getJSONObject("coordinates").getDouble("x"), 
+                        addressMatches.getJSONObject("coordinates").getDouble("y"));
+
+                    return Mono.just(new ResponseEntity<>("Verified Address:\n" + 
+                        formattedResponse.matchedAddress() + "\ncoordinates: \n" + formattedResponse.x() + ", " + formattedResponse.y(), HttpStatus.OK));
```

Comment:
Suggestion: i prefer my `Controller`s to be dumb. I think the extraction of the content should happen within the service itself.


Also, Spring should already have an instance of an `ObjectMapper` bean you could autowire in . This would eliminate the need for the additional JSON import.

Another Also, I think i would want this to return an object instead of a String. 

Ex: 
```
public Geolocation submitAddress(@RequestBody Address addres) {
...
```


I am not familiar with `Mono`, but at first glance it appears it sets up the request to be async to better handle multi threading, however, Spring Boot Controllers are multi-threaded by default. So you might be able to call `.block()` on the webclient request to retrieve the raw `String` instead of `Mono<String>`.